core/src/main/java/nl/nn/adapterframework/doc/IbisDocPipe.java

```diff
@@ -384,96 +385,153 @@ public PipeRunResult doPipe(Object input, IPipeLineSession session) throws PipeR
-		IbisDocExtractor extractor = new IbisDocExtractor();
-
-		// For all folders
-		for (String folder : groups.keySet()) {
-
-			// For all classes
-			for (IbisBean ibisBean : groups.get(folder)) {
-
-				// Copy the properties of FileSender into FilePipe so that the properties of FileHandler are also in FilePipe
-				Map<String, Method> beanProperties = getBeanProperties(ibisBean.getClazz());
-				if (beanProperties != null) {
-
-					if (ibisBean.getName().equals("FilePipe")) {
-						for (IbisBean bean : groups.get("Senders")) {
-							if (bean.getName().equals("FileSender")) {
-								Map<String, Method> senderProperties = getBeanProperties(bean.getClazz());
-								if (senderProperties != null) {
-									beanProperties.putAll(senderProperties);
-								}
-								break;
-							}
-						}
-					}
-
-					// For each method in the class
-					Iterator<String> iterator = new TreeSet<>(beanProperties.keySet()).iterator();
-
-					while (iterator.hasNext()) {
-
-						// Get the method
-						String property = iterator.next();
-						Method method = beanProperties.get(property);
-
-						// Get the ibisdoc of the method
-						IbisDoc ibisDoc = AnnotationUtils.findAnnotation(method, IbisDoc.class);
+	/**
+	 * Get the superclasses of a certain class.
+	 *
+	 * @param clazz - The class we have to derive the superclasses from
+	 * @return An ArrayList containing all the superclasses with priority given to them
+	 */
+	private ArrayList<String> getSuperClasses(Class clazz) {
+		ArrayList<String> superClasses = new ArrayList<String>();
+		int index = 0;
+		while (clazz.getSuperclass() != null) {
+
+			// Assign a string with a priority number attached to it and add it to the array of superclasses
+			String str = Integer.toString(index);
+			superClasses.add(clazz.getSimpleName() + str);
+			clazz = clazz.getSuperclass();
+			index++;
+		}
+		return superClasses;
+	}
+
+	/**
+	 * Gets the IbisDoc values.
+	 *
+	 * @param ibisDocValues - The String[] containing all the ibisDocValues
+	 * @return The needed ibisDocValues
+	 */
+	private String[] getValues(String[] ibisDocValues) {
+		int order;
+		int desc;
+		int def;
+
+		if (ibisDocValues[0].matches("\\d+")) {
+			order = Integer.parseInt(ibisDocValues[0]);
+			desc = 1;
+			def = 2;
+		} else {
+			order = 999;
+			desc = 0;
+			def = 1;
+		}
+		if (ibisDocValues.length > def)
+			return new String[]{ibisDocValues[desc], ibisDocValues[def], "" + order };
+		else
+			return new String[]{ibisDocValues[desc], "", "" + order };
+	}
+
+	/**
+	 * Add properties of the FilePipe to the FileSender.
+	 *
+	 * @param ibisBean       - The IbisBean that should be the FilePipe
+	 * @param groups         - Contains all information
+	 * @param beanProperties - The properties of a class (in this case the FilePipe)
+	 */
+	private void addPropertiesFileSender(IbisBean ibisBean, Map<String, TreeSet<IbisBean>> groups, Map<String, Method> beanProperties) {
+		if (ibisBean.getName().equals("FilePipe")) {
```

Comment:
Waarom is er hier een method om speciaal voor de FileSender de properties van de FilePipe te kopieren?
Er zijn dan toch veel meer senders die zo'n behandeling zouden willen?

Reply:
We hebben afgesproken dat dit in een volgende ronde aangepakt gaat worden, door middel van annotaties waarmee de instructie gegeven kan worden de ibisdoc op te halen van een method van een andere class